client/components/edit/datastream/DatastreamAgentsContent.test.tsx

```diff
@@ -0,0 +1,124 @@
+                setCurrentAgents: jest.fn(),
+                toggleDatastreamModal: jest.fn(),
+            },
+        };
+        datastreamOperationValues = {
+            uploadAgents: jest.fn(),
+            getAgents: jest.fn(),
+        };
+        mockUseEditorContext.mockReturnValue(editorValues);
+        mockUseDatastreamOperation.mockReturnValue(datastreamOperationValues);
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it("renders", () => {
+        const wrapper = shallow(<DatastreamAgentsContent />);
+        expect(toJson(wrapper)).toMatchSnapshot();
+    });
+
+    it("calls getAgents on render", async () => {
+        datastreamOperationValues.getAgents.mockResolvedValue([]);
+
+        const wrapper = mount(<DatastreamAgentsContent />);
+        await act(async () => {
+            await Promise.resolve(wrapper);
+            await new Promise((resolve) => setImmediate(resolve));
+            wrapper.update();
+        });
+
+        expect(datastreamOperationValues.getAgents).toHaveBeenCalled();
+        expect(editorValues.action.setCurrentAgents).toHaveBeenCalled();
+    });
+
+    it("saves current changes on save changes click", async () => {
+        datastreamOperationValues.getAgents.mockResolvedValue([]);
+
+        const wrapper = mount(<DatastreamAgentsContent />);
+        await act(async () => {
+            await Promise.resolve(wrapper);
+            await new Promise((resolve) => setImmediate(resolve));
+            wrapper.update();
+            wrapper.find("button.agentsSaveChangesButton").simulate("click");
+        });
+
+        expect(datastreamOperationValues.uploadAgents).toHaveBeenCalled();
+        expect(editorValues.action.toggleDatastreamModal).not.toHaveBeenCalled();
+    });
+
+    it("saves current agents and closes the modal", async () => {
+        datastreamOperationValues.getAgents.mockResolvedValue([]);
+
+        const wrapper = mount(<DatastreamAgentsContent />);
+        await act(async () => {
+            await Promise.resolve(wrapper);
+            await new Promise((resolve) => setImmediate(resolve));
+            wrapper.update();
+            wrapper.find("button.agentsSaveCloseButton").simulate("click");
+        });
+
+        expect(datastreamOperationValues.uploadAgents).toHaveBeenCalled();
+        expect(editorValues.action.toggleDatastreamModal).toHaveBeenCalled();
+    });
+
+    it("resets current agents on cancel", async () => {
+        datastreamOperationValues.getAgents.mockResolvedValue([]);
+
+        const wrapper = mount(<DatastreamAgentsContent />);
+        await act(async () => {
+            wrapper.find("button.agentsCancelButton").simulate("click");
+            wrapper.update();
+        });
+
+        expect(datastreamOperationValues.getAgents).toHaveBeenCalled();
+        expect(editorValues.action.setCurrentAgents).toHaveBeenCalled();
+        expect(editorValues.action.toggleDatastreamModal).toHaveBeenCalled();
```

Comment:
Would it make sense to assert that uploadAgents is not called in this context, just to be safe?

Reply:
Made change.